code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -300,6 +300,89 @@
 	icon_state = "gold"
 	item_state = "gold_id"
 
+/obj/item/weapon/card/id/nt_disguise
+	name = "\improper Nanotrasen undercover ID"
+	access = list(access_weapons, access_security, access_sec_doors, access_forensics_lockers, access_morgue, access_maint_tunnels, access_court, access_eva)
+	registered_name = null
+
+	var/registered_user = null
+	var/static/list/gimmick_names = list(
+		"A. N. Other",
+		"Guy Incognito",
+		"Hugh Zasking",
+		"Ivan Gottasecret"
+	)
+
+/obj/item/weapon/card/id/nt_disguise/attack_self(mob/user)
+
+	if(!src.registered_name)
+
+		if (ishuman(user))
+			var/mob/living/carbon/human/H = user
+			SetOwnerInfo(H)
+			alert(user,"Personal data gathered successfully; this includes: blood type, DNA, and fingerprints.\nYou may now proceed with the rest.","Nanotrasen undercover ID: notification","Ok")
+
+		var/n = input(user, "What name would you like to put on this card?", "Nanotrasen undercover ID: name") in gimmick_names
+		if(!n)
+			return
+		if (!Adjacent(user) || user.incapacitated())
+			return
+		src.registered_name = n
+
+		var/u = strict_ascii(sanitize(stripped_input(user, "What occupation would you like to put on this card?\nNote: this will not grant or remove any access levels.", "Nanotrasen undercover ID: occupation", "Detective", MAX_MESSAGE_LEN)))
+		if(!u)
+			alert("Invalid assignment.")
+			src.registered_name = null
+			return
+		if (!Adjacent(user) || user.incapacitated())
+			src.registered_name = null
+			return
+		src.assignment = u
+		src.name = "[src.registered_name]'s ID Card ([src.assignment])"
+		to_chat(user, "<span class='notice'>You successfully configured the NT ID card.</span>")
+
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			registered_user = H.dna.unique_enzymes
+
+	else if (!registered_user || user.dna.unique_enzymes && registered_user == user.dna.unique_enzymes)
```

Comment:
Remove the first occurrence of `.unique_enzymes` in this line. Otherwise the `&&` is redundant and it will runtime here if `user.dna` is null.